Sources/Purchasing/PurchasesOrchestrator.swift

```diff
@@ -584,36 +585,42 @@ private extension PurchasesOrchestrator {
 
             let completion = self.getAndRemovePurchaseCompletedCallback(forTransaction: transaction)
             let error = result.error
-            let nsError = error as NSError?
-            let finishable = (nsError?.userInfo[ErrorDetails.finishableKey as String] as? NSNumber)?.boolValue ?? false
+            let finishable = error?.finishable ?? false
```

Comment:
No more hacky introspecting of `userInfo` 🌮

Reply:
🪓